### DISCUSSION TITLE
What options do I have for creating a 1-D (line) nodal boundary on a 3D mesh?

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#26992] [Q&A Meshing]
I am using gmsh. For some reason, MOOSE does not recognize the "physical groups" created if they are 1-D (saying the sidesets do not exist). I tried using ExtraNodesetGenerator and manually listing a few points that should occur on the line, e.g

```
  [applied_force_disp]
    type = ExtraNodesetGenerator
	input = beam_mesh
	coord = '0 0 200; 0 12.5 200; 0 25 200; 0 -12.5 200; 0 -25 200'
	new_boundary = top-line
  []  
```

 but apparently MOOSE needs those exact points to exist in the mesh. What other options do I have for creating a 1-D boundary to apply a line load (as a displacement, i.e Dirichlet BC)?

**Comment by GiudGiud** [comment]
Hello

So it s a line load in a 3D mesh?
Can you use a nodeset instead of a sideset?
If the mesh is 3D you wont be able to make a sideset from points

Guillaume

**Reply by richmondodufisan** [reply]
Yes, it's a line load in a 3D mesh. So in gmsh I created a bunch of nodes, selected them, and named them as a physical group. By my understanding this should be a nodeset.

When I export the mesh as an msh file, moose does not even recognize the groups at all.
When I export it as an inp (abaqus) file, i get the error "expected boundary of dimension 2, got 1"

How else can I define these nodes?

**Reply by GiudGiud** [reply]
looking at the reader, 1D sidesets would be read as nodesets
https://mooseframework.inl.gov/docs/doxygen/libmesh/gmsh__io_8C_source.html

**Reply by GiudGiud** [reply]
in the test example in test/tests/meshgenerators/gmsh_bcs it looks like we can read nodesets from GMSH files

**Reply by richmondodufisan** [reply]
I'm not sure what the issue could be then- I tried creating both point and line groups and neither worked. Could you take a look and try to run it?

https://github.com/richmondodufisan/purple/tree/main/Gradient_Enhanced_Damage

**Reply by GiudGiud** [reply]
this is more of a gmsh question. @grmnptr do you remember who else is comfortable with gmsh?
